### PR TITLE
feat: Add `File.get_download_url()`

### PR DESCRIPTION
**Pull request by ArneGudermann**

### CODE REVIEW

File: src/viur/core/modules/file.py [hunk 1]
```diff
@@ -1161,6 +1161,56 @@ def add(self, skelType: SkelType, node: db.Key | int | str | None = None, *args,
...
+
+        if not self.canView("leaf", skel):
+            raise errors.Unauthorized()
+
+        dlkey = skel["dlkey"]
+
+        if derived and filename is None:
+            res = {}
+            for filename in skel["derived"]["files"]:
+                res[filename] = {"downloadUrl": self.create_download_url(dlkey, filename, derived)}
```
Comment (phorward) [inline src/viur/core/modules/file.py:1198]:
```suggestion
                res[filename] = self.create_download_url(dlkey, filename, derived)
```

File: src/viur/core/modules/file.py [hunk 2]
```diff
@@ -1161,6 +1161,56 @@ def add(self, skelType: SkelType, node: db.Key | int | str | None = None, *args,
...
+                # Check if Filename exist in the Derives. We sign nothing that not exist.
+                if filename not in skel["derived"]["files"]:
+                    raise errors.NotFound("File not in derives")
+            else:
+                if filename is None:
+                    filename = skel["name"]
+                elif filename != skel["name"]:
+                    raise errors.NotFound("Filename not match")
+
+            res = {"downloadUrl": self.create_download_url(dlkey, filename, derived)}
```
Comment (phorward) [inline src/viur/core/modules/file.py:1210]:
```suggestion
            res = self.create_download_url(dlkey, filename, derived)
```

**Review by phorward — CHANGES_REQUESTED** [review]
@ArneGudermann looks much better. Why this `{"downloadUrl": ...}`-thing everywhere, can we just use the url or the url per filename as suggested?

**Review by phorward — APPROVED** [review]
I'm fine with it!

**Review by sveneberth — APPROVED** [review]

**Pull request merged**